component/src/main/ballerina/redis/actions.bal

```diff
@@ -0,0 +1,1029 @@
+              string
+    }
+    public native function bitOpAnd(string destination, string[] keys)
+        returns (int|error);
+
+    documentation {
+        Perform bitwise OR between strings.
+
+        P{{destination}} Result key of the operation
+        P{{keys}} Input keys to perform AND between
+        R{{}} The size of the string stored in the destination key, that is equal to the size of the longest input
+              string
+    }
+    public native function bitOpOr(string destination, string[] keys)
+        returns (int|error);
+
+    documentation {
+        Perform bitwise NOT on a string.
+
+        P{{destination}} Result key of the operation
+        P{{key}} Input keys to perform AND between
+        R{{}} The size of the string stored in the destination key
+    }
+    public native function bitOpNot(string destination, string key)
+        returns (int|error);
+
+    documentation {
+        Perform bitwise XOR between strings.
+
+        P{{destination}} Result key of the operation
+        P{{key}} Input keys to perform AND between
+        R{{}} The size of the string stored in the destination key, that is equal to the size of the longest input string
+    }
+    public native function bitOpXor(string destination, string[] key)
+        returns (int|error);
+
+    documentation {
+        Decrement the integer value of a key by one.
+
+        P{{key}} The key
+        R{{}} The value of the key after the decrement
+    }
+    public native function decr(string key) returns (int|error);
+
+    documentation {
+        Decrement the integer value of a key by the given number.
+
+        P{{key}} The key
+        P{{value}} The value to be decremented
+        R{{}} The value of the key after the decrement
+    }
+    public native function decrBy(string key, int value) returns (int|error);
+
+    documentation {
+        Returns the bit value at offset in the string value stored at key.
+
+        P{{key}} The key
+        P{{offset}} The offset
+        R{{}} The bit value stored at offset
+    }
+    public native function getBit(string key, int offset) returns (int|error);
+
+    documentation {
+        Get a substring of the string stored at a key.
+
+        P{{key}} The key
+        P{{startPos}} The starting point of the substring
+        P{{end}} The end point of the substring
+        R{{}} The substring
+    }
+    public native function getRange(string key, int startPos, int end)
+        returns (string|error);
+
+    documentation {
+        Set the string value of a key and return its old value.
+
+        P{{key}} The key
+        P{{value}} The value to be set
+        R{{}} The old value stored at key, or null when key does not exist
+    }
+    public native function getSet(string key, string value) returns (string|error);
+
+    documentation {
+        Get the value of a key.
+
+        P{{key}} The key
+        R{{}} The value of the key, or null when key does not exist
```

Comment:
If this returning null make return type also to have nil. Check all null returning methods